ethereum/core/src/main/java/org/hyperledger/besu/ethereum/vm/EstimateGasOperationTracer.java

```diff
@@ -23,6 +23,8 @@
 
   private Gas sStoreStipendNeeded = Gas.ZERO;
 
+  private boolean isReverted = false;
```

Comment:
Instead of storing the revert flag in the tracer is it possible to use org.hyperledger.besu.ethereum.mainnet.TransactionProcessor.Result#getRevertReason?  (via org.hyperledger.besu.ethereum.transaction.TransactionSimulatorResult#getResult)?  If a TX reverts without a reason do we get an empty revert reason or a revert reason with zero bytes?

Reply:
Im thinking since this is only set by the revert operation and BLS precompiles that the optional will not be empty, at worst an optional of a zero length string

Reply:
this is what I tried to use at the beginning but by deploying a contract with this method I had empty revertReason `result.getResult().getRevertReason()` :

```
function store(uint256 num) public whenNotPaused {
        number = num;
        revert();
 }
```

Reply:
I just tried again and indeed we don't have an empty. I thought I found a specific case where it didn't work. but after doing it again it seems ok with your technique

Reply:
good catch. I just updated the code to use `getRevertReason`